// newest/lib/selectUpdates.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectUpdates = selectUpdates;

var _prompts = _interopRequireDefault(require("prompts"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

async function selectUpdates(currentVersions, updateVersions) {
  var _await$prompts$value;

  const updates = new Map();
  const selected = (_await$prompts$value = (await (0, _prompts.default)({
    type: "multiselect",
    name: "value",
    message: "Pick Updates",
    choices: [...updateVersions.keys()].map(name => {
      return {
        title: `${name} (${currentVersions.get(name)} -> ${updateVersions.get(name)})`,
        value: name
      };
    }),
    instructions: false,
    hint: "[space] to select, [enter] to submit, [a] for all"
  })).value) !== null && _await$prompts$value !== void 0 ? _await$prompts$value : [];

  for (const name of [...updateVersions.keys()]) {
    if (selected.includes(name)) {
      updates.set(name, updateVersions.get(name));
    }
  }

  return updates;
}
